connector/src/main/scala/com/basho/riak/spark/rdd/RiakRDD.scala

```diff
@@ -65,6 +65,13 @@ class RiakRDD[R] private[spark](@transient sc: SparkContext,
     partitions
   }
 
+  override def getPreferredLocations(split: Partition) : Seq[String] = {
+    split match {
+      case (lcp: RiakLocalCoveragePartition[_]) => List(lcp.primaryHost.getHost)
```

Comment:
I did this on purpose, to highlight that for that case there is no preferences at all. But, for sure we could call inherited method